var/da/da_radiance/da_qc_ahi.inc

```diff
@@ -422,12 +423,10 @@ subroutine da_qc_ahi (it, i, nchan, ob, iv)
          end do ! isuper
          end do ! jsuper
 
-         if ( iv%instid(i)%superob_width > 1 ) then
```

Comment:
iv%instid(i)%superob_width >= 1 always, I'm pretty sure.

In da_allocate_rad_iv.inc:
`iv%instid(i)%superob_width = 2*ahi_superob_halfwidth+1`

`ahi_superob_halfwidth` is the namelist variable.  In my testing I have set ahi_superob_halfwidth = 0, and it works just fine.

So, I think it's okay to remove the conditional.  Am I missing something?

Reply:
AHI superobbing was introduced in V4.2 with this PR https://github.com/wrf-model/WRF/pull/1173. I did not carefully check the logic of algorithm. You are right that if ahi_superob_halfwidth = 0, iv%instid(i)%superob_width=1 so no superobbing, in which case, the cloud_frac line will not be executed (condition is >1, not >=1). If we remove the condition, cloud_frac line will be executed anyway. In case of iv%instid(i)%superob_width=1, the denominator remains as 1, so I guess cloud_frac will remain as the original cloud mask (0 or 1). In conclusion, I think it is Ok to remove the condition.

Reply:
```
iv%instid(i)%cloud_frac(n) =  real( count(sum(clddet_tests,3) > 0),8) / real(iv%instid(i)%superob_width**2,8)
```
This line is a bit tricky to understand with 
```
integer*2          :: clddet_tests(iv%instid(i)%superob_width, &
                                      iv%instid(i)%superob_width, &
                                      num_clddet_tests)
```
Do you have idea what exactly it does? What is '8' for? I can guess it calculates cloud fraction from multiple pixels' cloud mask (0 or 1).

Reply:
Checking 'count' function, look like '8' is a KIND parameter, which means here using a 64 bit integer?

Reply:
Also, I think it's best to remove the condition about `iv%instid(i)%cloud_frac(n)`, because if running with no superobbing, `cloud_frac` will tell you on a per-pixel basis whether the pixel is clear or cloudy, which I think is useful.

Reply:
I agree.